container_api/src/container.rs

```diff
@@ -264,21 +253,22 @@ pub fn instantiate_from_config(
                 ))
             })?;
 
-            let network_config = instance_config
-                .network
-                .unwrap_or(default_network_config.to_owned())
-                .into();
+            let mut context_builder = ContextBuilder::new();
 
-            let context: Context = match instance_config.storage {
-                StorageConfiguration::File { path } => {
-                    create_file_context(&agent_config.id, &path, network_config)
-                        .map_err(|hc_err| format!("Error creating context: {}", hc_err.to_string()))
-                }
-                StorageConfiguration::Memory => {
-                    create_memory_context(&agent_config.id, network_config)
-                        .map_err(|hc_err| format!("Error creating context: {}", hc_err.to_string()))
-                }
-            }?;
+            let pub_key = KeyBuffer::with_corrected(&agent_config.public_address)?;
+            context_builder.with_agent(AgentId::new(&agent_config.name, &pub_key));
+
+            instance_config.network.map(|network_config| {
+                context_builder.with_network_config(JsonString::from(network_config))
+            });
+
+            if let StorageConfiguration::File { path } = instance_config.storage {
```

Comment:
where did the Memory context go?

Reply:
That's the default that `ContextBuilder` sets the context to if you just do a `ContextBuilder::new().spawn()`.